Kernel/Arch/i386/CPU.h

```diff
@@ -295,6 +295,8 @@ void load_task_register(u16 selector);
                                : "memory")
 #define sti() asm volatile("sti" :: \
                                : "memory")
+#define memory_barrier() asm volatile("" :: \
+                                          : "memory")
```

Comment:
The name sounds... misleading?
I was expecting `__builtin_ia32_mfence()` here tbh

Maybe name it `compiler_memory_barrier` instead?

Reply:
Maybe I need to do this in a better way, and I should use the `__builtin_ia32_mfence()` with the `compiler_memory_barrier` macro somehow...

Reply:
if I'm not mistaken, you don't _really_ need to stop the CPU from reordering this (there's no other nondependent memory ops around it anyway), you just want to stop the compiler from reordering it, so an `mfence` sounds pretty overkill - I think just naming the macro in a way that doesn't imply "this is an mfence" would be more than enough

Reply:
Well, I do also want to ensure that the CPU doesn't reorder on its own the access to memory, because the order of setting some of the registers in the HBA is very important.

Reply:
Hm didn't see this until now but generally for memory-mapped registers you just need to mark the fields in the structure that represents them as volatile. That ensures that the compiler generates code that reads/writes them in the specified order. But it doesn't imply a memory fence.